### PR TITLE
manifest: Update Matter revision to pull fixes for NCS 2.4.1

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
The newest version of sdk-connectedhomeip contains fixes that are needed for NCS 2.4.1 release.

### CODE REVIEW

File: samples/matter/lock/src/lock.matter [hunk 1]
```diff
@@ -1778,7 +1778,7 @@ endpoint 0 {
     callback attribute generatedCommandList;
     callback attribute acceptedCommandList;
     callback attribute attributeList;
-    ram      attribute featureMap default = 1;
+    ram      attribute featureMap default = 0;
```
Comment (Damian-Nordic) [inline samples/matter/lock/src/lock.matter:1781]: I think that this should be marked as "callback" aka "external" as the default value doesn't really mean anything.

**Review by Damian-Nordic — APPROVED** [review]

**Comment by LuDuda** [comment]
I was wondering if we should bump the revisions of applications, like here:

There are multiple of these, e.g. in the Matter Weather Station.
`CONFIG_MCUBOOT_IMAGE_VERSION="2.4.0+0"`

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -99,7 +99,10 @@ The Matter fork in the |NCS| (``sdk-connectedhomeip``) contains all commits from
 
 The following list summarizes the most important changes inherited from the upstream Matter:
 
-|no_changes_yet_note|
+* Fixed an IPC crash on nRF5340 when the Zephyr's Main thread takes a long time.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:102]:
```suggestion
* Fixed an IPC crash on nRF5340 when Zephyr's main thread takes a long time.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -99,7 +99,10 @@ The Matter fork in the |NCS| (``sdk-connectedhomeip``) contains all commits from
...
+* Added the :kconfig:option:`CHIP_MALLOC_SYS_HEAP_WATERMARKS_SUPPORT` Kconfig option to manage watermarks support.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:103]:
```suggestion
* Added the :kconfig:option:`CHIP_MALLOC_SYS_HEAP_WATERMARKS_SUPPORT` Kconfig option to manage watermark support.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -189,7 +192,10 @@ Thread samples
 Matter samples
 --------------
 
-|no_changes_yet_note|
+ * :ref:`matter_lock_sample` sample:
+
+   * Repaired the wrongly set feature map for software diagnostic cluster.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:197]:
```suggestion
   * Fixed the feature map for software diagnostic cluster, which was wrongly set.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 5]
```diff
@@ -189,7 +192,10 @@ Thread samples
...
+   * Repaired the wrongly set cluster revision for basic information cluster.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:198]:
```suggestion
   * Fixed the cluster revision for basic information cluster, which was wrongly set.
```

**Review by greg-fer — APPROVED** [review]

File: applications/matter_weather_station/configuration/thingy53_nrf5340_cpuapp/prj_release.conf [hunk 6]
```diff
@@ -58,7 +58,7 @@ CONFIG_AVERAGE_CURRENT_CONSUMPTION=10000
 CONFIG_BOARD_SERIAL_BACKEND_CDC_ACM=n
 
 # Bootloader and MCUmgr (OTA DFU) Configuration
-CONFIG_MCUBOOT_IMAGE_VERSION="2.4.0+0"
+CONFIG_MCUBOOT_IMAGE_VERSION="2.4.1+0"
```
Comment (LuDuda) [inline applications/matter_weather_station/configuration/thingy53_nrf5340_cpuapp/prj_release.conf:61]: Should we also find other occurrences of "2.4.0+0" and change it to "2.4.1+0" e.g. in Machine Learning sample.

**Review by LuDuda — APPROVED** [review]

File: applications/matter_weather_station/configuration/thingy53_nrf5340_cpuapp/prj_release.conf (hunk 6 above)
Comment (ArekBalysNordic) [reply applications/matter_weather_station/configuration/thingy53_nrf5340_cpuapp/prj_release.conf:61]: Can we do it in a separate PR?

Comment (LuDuda) [reply applications/matter_weather_station/configuration/thingy53_nrf5340_cpuapp/prj_release.conf:61]: Sure

**Pull request merged**